sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -664,8 +664,7 @@ private void validateAuthorizationGrantType(String registrationId, Authorization
                 + " is valid.", registrationId);
         }
 
-        if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
-            && AUTHORIZATION_CODE != grantType) {
+        if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId) && !AUTHORIZATION_CODE.equals(grantType)) {
```

Comment:
Shall we give a default value `authorization_code` to the grant type of the special oauth2 client(azure)?  this validation only works when the user overrides the grant type, or else the default value will be used.

Reply:
Actually, the configured one will not be used. 
Here is related code about `azure` client's grant type:
https://github.com/Azure/azure-sdk-for-java/blob/spring-cloud-azure-autoconfigure_4.3.0/sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadClientRegistrationRepository.java#L83-L85

![image](https://user-images.githubusercontent.com/13167207/184048363-cd858c04-5967-48d8-b6cf-4db598355c18.png)


Here is just a valication.